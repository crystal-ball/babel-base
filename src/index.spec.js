jest.mock('path')

import { babelBase } from '.'

describe('babelBase', () => {
  test('when called with target node, then node babel configs are returned', () => {
    expect(babelBase({ env: 'development', target: 'node' })).toMatchSnapshot()
  })

  test('when called with target react for dev, then development react configs are returned', () => {
    expect(babelBase({ env: 'development', target: 'react' })).toMatchSnapshot()
  })

  test('when called with target react for test, then test react configs are returned', () => {
    expect(babelBase({ env: 'test', target: 'react' })).toMatchSnapshot()
  })

  test('when called with target react for prod, then production react configs are returned', () => {
    expect(babelBase({ env: 'production', target: 'react' })).toMatchSnapshot()
  })
})
