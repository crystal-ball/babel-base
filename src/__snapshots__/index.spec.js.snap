// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babelBase when called with target node, then node babel configs are returned 1`] = `
Object {
  "plugins": Array [
    Array [
      "@babel/plugin-transform-runtime",
      Object {
        "useESModules": false,
        "version": "7.10.1",
      },
    ],
    Array [
      "babel-plugin-transform-import-aliases",
      Object {
        "aliases": Object {
          "@": "/usr/code/src",
        },
      },
    ],
    "@babel/plugin-proposal-class-properties",
  ],
  "presets": Array [
    Array [
      "@babel/preset-env",
      Object {
        "corejs": 3,
        "modules": "commonjs",
        "targets": "node 12",
        "useBuiltIns": "entry",
      },
    ],
    "@babel/preset-typescript",
  ],
}
`;

exports[`babelBase when called with target react for dev, then development react configs are returned 1`] = `
Object {
  "env": Object {
    "production": Object {
      "plugins": Array [
        "transform-react-remove-prop-types",
      ],
    },
  },
  "plugins": Array [
    "react-hot-loader/babel",
    Array [
      "@babel/plugin-transform-runtime",
      Object {
        "useESModules": true,
        "version": "7.10.1",
      },
    ],
    Array [
      "babel-plugin-transform-import-aliases",
      Object {
        "aliases": Object {
          "@": "/usr/code/src",
        },
      },
    ],
    "@babel/plugin-proposal-class-properties",
    "@babel/plugin-proposal-private-methods",
  ],
  "presets": Array [
    Array [
      "@babel/preset-env",
      Object {
        "corejs": 3,
        "modules": false,
        "targets": "defaults",
        "useBuiltIns": "entry",
      },
    ],
    Array [
      "@babel/preset-react",
      Object {
        "development": true,
        "useBuiltIns": true,
      },
    ],
    "@babel/preset-typescript",
  ],
}
`;

exports[`babelBase when called with target react for prod, then production react configs are returned 1`] = `
Object {
  "env": Object {
    "production": Object {
      "plugins": Array [
        "transform-react-remove-prop-types",
      ],
    },
  },
  "plugins": Array [
    "react-hot-loader/babel",
    Array [
      "@babel/plugin-transform-runtime",
      Object {
        "useESModules": true,
        "version": "7.10.1",
      },
    ],
    Array [
      "babel-plugin-transform-import-aliases",
      Object {
        "aliases": Object {
          "@": "/usr/code/src",
        },
      },
    ],
    "@babel/plugin-proposal-class-properties",
    "@babel/plugin-proposal-private-methods",
  ],
  "presets": Array [
    Array [
      "@babel/preset-env",
      Object {
        "corejs": 3,
        "modules": false,
        "targets": "defaults",
        "useBuiltIns": "entry",
      },
    ],
    Array [
      "@babel/preset-react",
      Object {
        "development": false,
        "useBuiltIns": true,
      },
    ],
    "@babel/preset-typescript",
  ],
}
`;

exports[`babelBase when called with target react for test, then test react configs are returned 1`] = `
Object {
  "env": Object {
    "production": Object {
      "plugins": Array [
        "transform-react-remove-prop-types",
      ],
    },
  },
  "plugins": Array [
    "react-hot-loader/babel",
    Array [
      "@babel/plugin-transform-runtime",
      Object {
        "useESModules": false,
        "version": "7.10.1",
      },
    ],
    Array [
      "babel-plugin-transform-import-aliases",
      Object {
        "aliases": Object {
          "@": "/usr/code/src",
        },
      },
    ],
    "@babel/plugin-proposal-class-properties",
    "@babel/plugin-proposal-private-methods",
  ],
  "presets": Array [
    Array [
      "@babel/preset-env",
      Object {
        "corejs": 3,
        "modules": "commonjs",
        "targets": "defaults",
        "useBuiltIns": "entry",
      },
    ],
    Array [
      "@babel/preset-react",
      Object {
        "development": false,
        "useBuiltIns": true,
      },
    ],
    "@babel/preset-typescript",
  ],
}
`;
